bin/pub

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+# Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+# for details. All rights reserved. Use of this source code is governed by a
+# BSD-style license that can be found in the LICENSE file.
+
+# Runs ./pub.dart with dart from PATH
+
+dart `dirname "$0"`/pub.dart $*
```

Comment:
Why this file?

Reply:
This file is invoked by the binstub in the tests.
getEnvironment() now returns a path with bin/ in front, so these are invoked instead of pub from the sdk. (See change to  `test/global/binstubs/utils.dart`)

Reply:
I think `test/bin` or `tool/test-bin/` might be better...

`bin/` has a special meaning in the pub world.

Reply:
👍